sky/backends/cloud_vm_ray_backend.py

```diff
@@ -4372,24 +4374,25 @@ def post_teardown_cleanup(self,
             # The cluster yaml does not exist when skypilot has not found
             # the right resource to provision the cluster.
             if handle.cluster_yaml is not None:
+                launched_resources = (
+                    handle.launched_resources.assert_launchable())
+                cloud = launched_resources.cloud
+                config = common_utils.read_yaml(handle.cluster_yaml)
```

Comment:
This could throw if the cluster yaml is missing or invalid. If `purge` is true, we should not completely crash out.

Reply:
The `handle.cluster_yaml` is checked in line 4376, so it should be OK here?